srcs/app/config/analyser/Token.cpp

```diff
@@ -0,0 +1,94 @@
+#include <parser/Token.hpp>
+
+std::string Token::Type::GetTokenTypeStr(enum e_id type) {
+	std::string retval("unknown");
+	if (type > T_UNKNOWN && type < T_INVALID)
+		return str_map[type];
+	return retval;
+}
+
+const char *Token::Type::str_map[10] = {
+	"t scope open",
+	"t scope close",
+	"t symbol",
+	"t str immediate t0",
+	"t str immediate t1",
+	"t int immediate",
+	"t uint immediate",
+	"t dbl immediate",
+	"t end"
+};
+
+std::string Token::State::GetParsingStateTypeStr(enum e_id type) {
+	std::string retval("");
+	if (type > K_NONE && type < K_LAST_INVALID_STATE)
+		return keyword_to_str[type];
+	return retval;
+}
+
+t_parsing_state Token::State::GetParsingStateTypeEnum
+(const std::string &data, t_token_type ttype) {
+	if (ttype == Token::Type::T_END || ttype == Token::Type::T_SCOPE_OPEN
+		|| ttype == Token::Type::T_SCOPE_CLOSE)
+		return K_EXP_KW;
+	for (uint8_t i = K_EXIT; i < K_LAST_INVALID_STATE; ++i) {
+		if (keyword_to_str[i + 1] == data)
+			return static_cast<t_parsing_state>(i);
+	}
+	return K_NONE;
+}
+
+const char *Token::State::keyword_to_str[20] = {
+	"",
+	"EXIT",
+	"INIT",
+	"server",
+	"listen",
+	"server_name",
+	"root",
+	"client_max_body_size",
+	"error_page",
+	"return",
+	"autoindex",
+	"index",
+	"upload_store",
+	"return_status",
```

Comment:
I'm not sure how should this look like, it's not in http://nginx.org/en/docs/http/ngx_http_core_module.html
Is it just "return" directive?

Reply:
Yes, this is the [return](https://nginx.org/en/docs/http/ngx_http_rewrite_module.html#return) directive. In nginx, it can appear in both server and location, and it could have multiple syntaxes:

> **return** _code [text];_
**return** _code URL;_
**return** _URL;_

For simplicity, we could support only the second version (**return** _code URL_).

The `return_status` variable from CommonConfig will store the code status, and `return_url` the return URL.
We don't have to add them to the list of directives since we already have the return directive in there.